pantheon/src/main/java/tech/pegasys/pantheon/cli/subcommands/blocks/BlocksSubCommand.java

```diff
@@ -241,71 +241,127 @@ private MiningParameters getMiningParameters() {
       } finally {
         metricsService.ifPresent(MetricsService::stop);
       }
     }
 
-    private static void checkCommand(
+    private PantheonController<?> createPantheonController() {
+      return parentCommand.parentCommand.buildController();
+    }
+
+    private void exportRlpFormat(final PantheonController<?> controller) throws IOException {
+      final ProtocolContext<?> context = controller.getProtocolContext();
+      RlpBlockExporter exporter =
+          parentCommand.rlpBlockExporterFactory.get(context.getBlockchain());
+      exporter.exportBlocks(blocksExportFile, getStartBlock(), getEndBlock());
+    }
+
+    private void checkCommand(
         final ExportSubCommand exportSubCommand, final Long startBlock, final Long endBlock) {
       checkNotNull(exportSubCommand.parentCommand);
-      checkNotNull(exportSubCommand.parentCommand.blockExporter);
-      checkNotNull(startBlock);
-      if (startBlock < 0) {
-        throw new CommandLine.ParameterException(
-            new CommandLine(exportSubCommand),
-            "--start-block must be greater than or equal to zero");
+
+      final Optional<Long> maybeStartBlock = getStartBlock();
+      final Optional<Long> maybeEndBlock = getEndBlock();
+
+      maybeStartBlock
+          .filter(blockNum -> blockNum < 0)
+          .ifPresent(
+              (blockNum) -> {
+                throw new CommandLine.ParameterException(
+                    spec.commandLine(),
+                    "Parameter --start-block ("
+                        + blockNum
+                        + ") must be greater than or equal to zero.");
+              });
+
+      maybeEndBlock
+          .filter(blockNum -> blockNum < 0)
+          .ifPresent(
+              (blockNum) -> {
+                throw new CommandLine.ParameterException(
+                    spec.commandLine(),
+                    "Parameter --end-block ("
+                        + blockNum
+                        + ") must be greater than or equal to zero.");
+              });
+
+      if (maybeStartBlock.isPresent() && maybeEndBlock.isPresent()) {
+        if (endBlock <= startBlock) {
+          throw new CommandLine.ParameterException(
+              spec.commandLine(),
+              "Parameter --end-block ("
+                  + endBlock
+                  + ") must be greater start block ("
+                  + startBlock
+                  + ").");
+        }
       }
-      if (endBlock != null && startBlock >= endBlock) {
+
+      // Error if data directory is empty
+      Path databasePath =
+          Paths.get(
+              parentCommand.parentCommand.dataDir().toAbsolutePath().toString(),
+              PantheonController.DATABASE_PATH);
+      File databaseDirectory = new File(databasePath.toString());
+      if (!databaseDirectory.exists()
+          || !databaseDirectory.isDirectory()
+          || databaseDirectory.list().length == 0) {
```

Comment:
`isDirectory` implies `exists` so `exists` can be removed.

Reply:
I don't think this is more readable but just for fun, the whole thing can be determined from `list()` because it returns `null` if it's not a directory or doesn't exist :P 
`Optional.ofNullable(databaseDirectory.list()).filter(files -> files.length == 0).isPresent()`